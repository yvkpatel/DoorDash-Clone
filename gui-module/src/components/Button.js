const Button = ({text, color, onClick}) => {
    return(
        <button className='button'
        style={{background:color}}
        data-testid="button"
        onClick={onClick}>
            {text}
        </button>
    )
}

export default Button