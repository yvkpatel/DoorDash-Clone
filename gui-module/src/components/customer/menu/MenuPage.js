/* Displays menu items available at the selected restaurant*/

import { useState } from "react";
import MenuItem from "./MenuItem";
import Header from "../../Header";
import MenuList from "./MenuList";
import Cart from "./Cart";
import Button from "../../Button";
import { Link } from "react-router-dom";

const MenuPage = (props) => {
  const cart = new Cart();

  // Retrieving data from previous page (Customer's Restaurant List)
  var restaurantId; // @JR use this id for fetching the list below.

  const st = props.location.state;
  if (st != null) {
    restaurantId = st.restaurantId;
  }

  // This list would be pulled from databases.
  const [items /*, setTasks*/] = useState([
    {
      id: 1,
      title: "Pizza",
      price: 3.55,
      image:
        "https://img.cdn4dd.com/cdn-cgi/image/fit=cover,width=300,height=300,format=jpeg,quality=50/https://doordash-static.s3.amazonaws.com/media/photos/f71cc006-15c5-40be-bfac-91cdfdb0adb8-retina-large-jpeg",
      details:
        "This pizza has been finely topped with chedder cheese over a 100% pulverized tomato base on our signature creamy pizza dough.",
    },
    {
      id: 2,
      title: "Pizza 2",
      price: 6.55,
      image:
        "https://img.cdn4dd.com/cdn-cgi/image/fit=cover,width=300,height=300,format=jpeg,quality=50/https://doordash-static.s3.amazonaws.com/media/photos/f71cc006-15c5-40be-bfac-91cdfdb0adb8-retina-large-jpeg",
      details:
        "This pizza has been finely topped with chedder cheese over a 100% pulverized tomato base on our signature creamy pizza dough.",
    },
    {
      id: 3,
      title: "Pizza 3",
      price: 5.55,
      image:
        "https://img.cdn4dd.com/cdn-cgi/image/fit=cover,width=300,height=300,format=jpeg,quality=50/https://doordash-static.s3.amazonaws.com/media/photos/f71cc006-15c5-40be-bfac-91cdfdb0adb8-retina-large-jpeg",
      details:
        "This pizza has been finely topped with chedder cheese over a 100% pulverized tomato base on our signature creamy pizza dough.",
    },
  ]);

  return (
    <div className="App" data-testid="menu-page">
      <Header />
      <div /*style={{ width: "75%" }}*/>
        <MenuList itemList={items} component={MenuItem} cart={cart} />
      </div>
      <Link to={{ pathname: "../payment", state: { cart: cart } }}>
        <Button text={"Checkout"} color={"#2DB6F0"} />
      </Link>
    </div>
  ); // @JR Link here, sends to payment page. Note: Currently the link doesn't care is cart is empty.
};

export default MenuPage;
