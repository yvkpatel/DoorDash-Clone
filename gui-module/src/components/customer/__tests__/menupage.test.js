import { render, screen, cleanup, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import userEvent from "@testing-library/user-event";
import React, { useEffect } from "react";
import { BrowserRouter } from "react-router-dom";
import ReactDOM from "react-dom";
import Button from "../../Button";
import MenuPage from "../menu/MenuPage";
import "@testing-library/jest-dom/extend-expect";

afterEach(() => {
  cleanup();
});

const MockProp = {
  location: {
    state: {
      restaurantId: 1,
    },
  },
};

test("should render customer menu component", () => {
  render(
    <BrowserRouter>
      <MenuPage {...MockProp} />
    </BrowserRouter>
  );
  const MenuPageElement = screen.getByTestId("menu-page");
  expect(MenuPageElement).toBeDefined();
});

test("Add to Cart button renders", () => {
  render(
    <BrowserRouter>
      <MenuPage {...MockProp} />
    </BrowserRouter>
  );
  const CartButtonElement = screen.getAllByText("Add to Cart");
  expect(CartButtonElement).toBeDefined();
});

test("Add to Cart buttons are enabled", () => {
  render(
    <BrowserRouter>
      <MenuPage {...MockProp} />
    </BrowserRouter>
  );
  const CartButtonElements = screen.getAllByRole("button", {
    name: "Add to Cart",
  });

  CartButtonElements.forEach((element) => {
    expect(element).toBeEnabled();
  });
});

test("clicking checkout should go to the payments page", () => {
  render(
    <BrowserRouter>
      <MenuPage {...MockProp} />
    </BrowserRouter>
  );
  fireEvent.click(screen.getByText("Checkout"));
  const newURL = window.location.href;
  expect(newURL).toBe("http://localhost/payment");
});

test("matches snapshot", () => {
  const tree = renderer.create(
    <BrowserRouter>
      <MenuPage {...MockProp} />
    </BrowserRouter>
  );
  expect(tree).toMatchSnapshot();
});
