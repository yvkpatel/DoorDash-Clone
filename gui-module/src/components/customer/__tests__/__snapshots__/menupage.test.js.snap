// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="App"
  data-testid="menu-page"
>
  <header
    className="header"
  >
    <img
      alt="YUMI Logo"
      className="logo"
      src="logo-header.png"
    />
    <a
      className="accSettings"
      href="https://developer.mozilla.org/en-US/docs/Web/CSS"
    >
      Account Settings
    </a>
    <a
      href="/"
      onClick={[Function]}
    >
      <button
        className="button"
        data-testid="button"
        onClick={[Function]}
        style={
          Object {
            "background": "#2DB6F0",
          }
        }
      >
        Logout
      </button>
    </a>
  </header>
  <div>
    <div
      className="menulist"
    >
      <div
        className="testCard"
        style={
          Object {
            "display": "flex",
            "width": "100%",
          }
        }
      >
        <img
          alt="foodItem"
          className="thumbnail"
          src="https://img.cdn4dd.com/cdn-cgi/image/fit=cover,width=300,height=300,format=jpeg,quality=50/https://doordash-static.s3.amazonaws.com/media/photos/f71cc006-15c5-40be-bfac-91cdfdb0adb8-retina-large-jpeg"
        />
        <div
          className="details"
        >
          <h1
            className="md-1"
          >
            Pizza
          </h1>
          <hr
            style={
              Object {
                "marginBottom": "2px",
                "marginTop": "2px",
                "width": "100%",
              }
            }
          />
          <p
            style={
              Object {
                "margin": "0px",
                "verticalAlign": "top",
              }
            }
          >
            This pizza has been finely topped with chedder cheese over a 100% pulverized tomato base on our signature creamy pizza dough.
          </p>
          <p
            style={
              Object {
                "float": "right",
                "marginBottom": "0px",
                "marginTop": "16px",
                "verticalAlign": "top",
              }
            }
          >
            $
            3.55
             CAD
          </p>
        </div>
        <div
          className="add-to-cart"
        >
          <button
            className="button"
            data-testid="button"
            onClick={[Function]}
            style={
              Object {
                "background": "#2DB6F0",
              }
            }
          >
            Add to Cart
          </button>
          <p>
            Quantity:
          </p>
          <p>
            1
          </p>
        </div>
      </div>
      <div
        className="testCard"
        style={
          Object {
            "display": "flex",
            "width": "100%",
          }
        }
      >
        <img
          alt="foodItem"
          className="thumbnail"
          src="https://img.cdn4dd.com/cdn-cgi/image/fit=cover,width=300,height=300,format=jpeg,quality=50/https://doordash-static.s3.amazonaws.com/media/photos/f71cc006-15c5-40be-bfac-91cdfdb0adb8-retina-large-jpeg"
        />
        <div
          className="details"
        >
          <h1
            className="md-1"
          >
            Pizza 2
          </h1>
          <hr
            style={
              Object {
                "marginBottom": "2px",
                "marginTop": "2px",
                "width": "100%",
              }
            }
          />
          <p
            style={
              Object {
                "margin": "0px",
                "verticalAlign": "top",
              }
            }
          >
            This pizza has been finely topped with chedder cheese over a 100% pulverized tomato base on our signature creamy pizza dough.
          </p>
          <p
            style={
              Object {
                "float": "right",
                "marginBottom": "0px",
                "marginTop": "16px",
                "verticalAlign": "top",
              }
            }
          >
            $
            6.55
             CAD
          </p>
        </div>
        <div
          className="add-to-cart"
        >
          <button
            className="button"
            data-testid="button"
            onClick={[Function]}
            style={
              Object {
                "background": "#2DB6F0",
              }
            }
          >
            Add to Cart
          </button>
          <p>
            Quantity:
          </p>
          <p>
            1
          </p>
        </div>
      </div>
      <div
        className="testCard"
        style={
          Object {
            "display": "flex",
            "width": "100%",
          }
        }
      >
        <img
          alt="foodItem"
          className="thumbnail"
          src="https://img.cdn4dd.com/cdn-cgi/image/fit=cover,width=300,height=300,format=jpeg,quality=50/https://doordash-static.s3.amazonaws.com/media/photos/f71cc006-15c5-40be-bfac-91cdfdb0adb8-retina-large-jpeg"
        />
        <div
          className="details"
        >
          <h1
            className="md-1"
          >
            Pizza 3
          </h1>
          <hr
            style={
              Object {
                "marginBottom": "2px",
                "marginTop": "2px",
                "width": "100%",
              }
            }
          />
          <p
            style={
              Object {
                "margin": "0px",
                "verticalAlign": "top",
              }
            }
          >
            This pizza has been finely topped with chedder cheese over a 100% pulverized tomato base on our signature creamy pizza dough.
          </p>
          <p
            style={
              Object {
                "float": "right",
                "marginBottom": "0px",
                "marginTop": "16px",
                "verticalAlign": "top",
              }
            }
          >
            $
            5.55
             CAD
          </p>
        </div>
        <div
          className="add-to-cart"
        >
          <button
            className="button"
            data-testid="button"
            onClick={[Function]}
            style={
              Object {
                "background": "#2DB6F0",
              }
            }
          >
            Add to Cart
          </button>
          <p>
            Quantity:
          </p>
          <p>
            1
          </p>
        </div>
      </div>
    </div>
  </div>
  <a
    href="../payment"
    onClick={[Function]}
  >
    <button
      className="button"
      data-testid="button"
      style={
        Object {
          "background": "#2DB6F0",
        }
      }
    >
      Checkout
    </button>
  </a>
</div>
`;
